cli/azd/pkg/project/project.go

```diff
@@ -271,7 +299,28 @@ func Save(ctx context.Context, projectConfig *ProjectConfig, projectFilePath str
 		return fmt.Errorf("saving project file: %w", err)
 	}
 
-	projectConfig.Path = projectFilePath
+	projectConfig.Path = filepath.Dir(projectFilePath)
 
 	return nil
 }
+
+// hooksFromInfraModule check if there is file named azd.hooks.yaml in the service path
+// and return the hooks configuration.
+func hooksFromInfraModule(infraPath, moduleName string) (HooksConfig, error) {
+	hooksPath := filepath.Join(infraPath, moduleName+".hooks.yaml")
+	if _, err := os.Stat(hooksPath); os.IsNotExist(err) {
+		return nil, nil
+	}
+	hooksFile, err := os.ReadFile(hooksPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading hooks from '%s', %w", hooksPath, err)
+	}
+
+	// open hooksPath into a byte array and unmarshal it into a map[string]*ext.HookConfig
+	hooks := make(HooksConfig)
```

Comment:
Would `var hooks HooksConfig` have worked here?

Reply:
I need to init the `hooks`, so either make or `hooks:=HooksConfig{}`